File: Leetcode/Easy/twoSum.py
'''
Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.
'''
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for index, number in enumerate(nums):
            need = target - number
            if need not in seen:
                seen[number] = index
            elif need in seen:
                return [seen.get(need), index]

File: Leetcode/Easy/twoSum.cpp
/*
Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.
*/
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); ++i){
            int need = target - nums[i];
            if(seen.find(need) == seen.end()){
                seen[nums[i]] = i;
            } else{
                return std::vector<int> {seen.at(need), i};
            }
        }
        return std::vector<int>();
    }
};